expression-language.md

```diff
@@ -0,0 +1,369 @@
+case an error arise, the evaluator collects a list of errors, referred in this spec as _error list_, which is then
+returned together with the evaluated value of the CESQL expression.
+
+Whenever possible, some errors checks SHOULD be done at compile time by the expression evaluator, in order to prevent
+runtime errors.
+
+### 3.4. Operators
+
+The following tables show the operators that MUST be supported by a CESQL evaluator.
+
+All the operators in the following tables are listed in precedence order.
+
+#### 3.4.1. Unary operators
+
+Corresponds to the syntactic rule `unary-operation`:
+
+| Definition                  | Semantics                       |
+| --------------------------- | ------------------------------- |
+| `NOT x: Boolean -> Boolean` | Returns the negate value of `x` |
+| `-x: Integer -> Integer`    | Returns the minus value of `x`  |
+
+#### 3.4.2. Binary operators
+
+Corresponds to the syntactic rule `binary-operation`:
+
+| Definition                              | Semantics                                                                                       |
+| --------------------------------------- | ----------------------------------------------------------------------------------------------- |
+| `x = y: Boolean x Boolean -> Boolean`   | Returns `true` if the values of `x` and `y` are equal                                           |
+| `x AND y: Boolean x Boolean -> Boolean` | Returns the logical and of `x` and `y`                                                          |
+| `x OR y: Boolean x Boolean -> Boolean`  | Returns the logical or of `x` and `y`                                                           |
+| `x XOR y: Boolean x Boolean -> Boolean` | Returns the logical xor of `x` and `y`                                                          |
+| `x = y: Integer x Integer -> Boolean`   | Returns `true` if the values of `x` and `y` are equal                                           |
+| `x < y: Integer x Integer -> Boolean`   | Returns `true` if `x` is strictly lower than `y`                                                |
+| `x <= y: Integer x Integer -> Boolean`  | Returns `true` if `x` is lower or equal to `y`                                                  |
+| `x > y: Integer x Integer -> Boolean`   | Returns `true` if `x` is strictly greater than `y`                                              |
+| `x >= y: Integer x Integer -> Boolean`  | Returns `true` if `x` is greater or equal to `y`                                                |
+| `x * y: Integer x Integer -> Integer`   | Returns the product of `x` and `y`                                                              |
+| `x / y: Integer x Integer -> Integer`   | Returns the rounded division of `x` and `y`. Returns `0` if `y = 0` and raise an error          |
+| `x % y: Integer x Integer -> Integer`   | Returns the remainder of the division of `x` and `y`. Returns `0` if `y = 0` and raise an error |
```

Comment:
For clarity it might be useful to also add example where lhs is negative. Sometimes reminder/modulo get confused. For instance in `C` and `Python` the `%` operator have different semantic.

Reply:
> rounded division

does that mean:
* truncated division -- rounded to zero
* floored division -- rounded downward
* rounded to nearest integer

https://en.wikipedia.org/wiki/Modulo_operation

Reply:
I would picked `truncated division` since it seems that what `SQLite`, `Go` and `Clojure` does. Also `%` should behave accordingly.

@clemensv this is probably related to your rounding comment.

Reply:
check it out now

Reply:
I think it's good now.